packages/datatrak-web/src/features/Questions/GeolocateQuestion/GeolocateQuestion.tsx

```diff
@@ -74,20 +114,28 @@ export const GeolocateQuestion = ({
 
         {displayMapModalButton && (
           <>
-            <SeparatorText>or</SeparatorText>
-            <ModalButton onClick={toggleMapModal}>
-              <MapIcon />
-              <ButtonText>Drop pin on map</ButtonText>
-            </ModalButton>
+            <OrDivider />
+            <StyledButton
+              onClick={shouldUseDetectPosition ? populateFromCurrentPosition : toggleMapModal}
+              fullWidth={isMobile}
+              variant={isMobile ? 'contained' : 'text'}
+              startIcon={shouldUseDetectPosition ? <LocateIcon /> : <MapIcon />}
+            >
+              {shouldUseDetectPosition ? 'Detect current location' : 'Drop pin on map'}
+            </StyledButton>
```

Comment:
_Optional_: This looks a bit dense to me, do you reckon we should extract that into a separate component like `GeolocationActions`?

And then here just {displayMapModalButton && <GeolocationActions />}